core/src/main/java/org/keycloak/sdjwt/consumer/JwtVcMetadataTrustedSdJwtIssuer.java

```
@@ -0,0 +1,190 @@
+    }
+
+    @Override
+    public List<SignatureVerifierContext> resolveIssuerVerifyingKeys(IssuerSignedJWT issuerSignedJWT)
+            throws VerificationException {
+        // Read iss (claim) and kid (header)
+        String iss = Optional.ofNullable(issuerSignedJWT.getPayload().get("iss"))
+                .map(JsonNode::asText)
+                .orElse("");
+        String kid = issuerSignedJWT.getHeader().getKeyId();
+
+        // Match the read iss claim against the trusted pattern
+        Matcher matcher = issuerUriPattern.matcher(iss);
+        if (!matcher.matches()) {
+            throw new VerificationException(String.format(
+                    "Unexpected Issuer URI claim. Expected=/%s/, Got=%s",
+                    issuerUriPattern.pattern(), iss
+            ));
+        }
+
+        // As per specs, only HTTPS URIs are supported
+        validateHttpsIssuerUri(iss);
+
+        // Fetch and collect exposed JWKs
+        List<JsonNode> jwks = new ArrayList<>();
+        for (JsonNode jwk : fetchIssuerMetadata(iss)) {
+            jwks.add(jwk);
+        }
+
+        // If kid specified, only consider the first matching key
+        if (kid != null) {
+            jwks = jwks.stream()
+                    .filter(jwk -> {
+                        JsonNode jwkKid = jwk.get("kid");
+                        return jwkKid != null && jwkKid.asText().equals(kid);
+                    })
+                    .findFirst()
+                    .map(Collections::singletonList)
+                    .orElse(Collections.emptyList());
+        }
+
+        // Build JWSVerifier's
+        List<SignatureVerifierContext> verifiers = new ArrayList<>();
+        for (JsonNode jwk : jwks) {
+            try {
+                verifiers.add(JwkParsingUtils.convertJwkToVerifierContext(jwk));
+            } catch (Exception e) {
+                throw new VerificationException("A potential JWK was retrieved but found invalid");
```

This swallows the intial parse exception, which could be helpful for debugging. Maybe add it back to the verfication exception or log it on debug?

---

Thank you. I added it back to the verification exception.